### ISSUE TITLE
src/kafka.c: enable.auto.commit property problems

### ISSUE THREAD

**Issue by 65278** [adjust/schaufel#93]
**Problem**:
When applied to a topic, `enable.auto.commit` throws a deprecation error as it is an alias for `auto.commit.enable`.
To forego deprecation, I've moved the consumer offset commit from per topic to per config (global) in this commit:
https://github.com/adjust/schaufel/commit/3367570256c7b1b70c507dfbff16bd06bdf57d18

In production, this had a serious side effect. Kafka servers would rebalance consumer groups constantly because consumers would fail to commit offsets and therefore time out. This was due to overload of kafka when commiting offsets.

This happened because the old config was inconsistent. This is how the relevant properties look like right before spawning a new consumer:
```
(gdb) info line
Line 2305 of "rdkafka.c" starts at address 0x7ffff7df45ba <rd_kafka_new+2634> and ends at 0x7ffff7df45c9 <rd_kafka_new+2649>.
(gdb) p app_conf->topic_conf->auto_commit_interval_ms
$37 = 10
(gdb) p app_conf->auto_commit_interval_ms
$38 = 5000
(gdb) p app_conf->enable_auto_commit
$39 = 1
(gdb) p app_conf->topic_conf->auto_commit
$40 = 1
```

Both `app_conf->topic_conf->auto_commit` (the old style consumer that throws a warning) and `app_conf->enable_auto_commit` are set to `true`. Subsequently, librdkafka never starts a legacy consumer but a new style consumer. Unfortunately, `auto.commit.interval.ms` was configured for the old style consumer only. Librdkafka still throws a warning.
This happens because `enable.auto.commit` on a global level is a default setting in librdkafka.

When moving the consumer over to the new style consumer, `auto.commit.interval.ms` changed from the default 5000ms to the envisioned 10ms. This proved to be too much for our kafka servers.

TL;DR: our consumers never worked as intended and a change unmasked this logic error by breaking production.
